### ISSUE TITLE
Rectangle missing flip use case from deprecated shape Quad

### ISSUE THREAD

**Issue by tripokey** [bevyengine/bevy#11996]
## What problem does this solve or what need does it fill?

The deprecated shape Quad's flip field does not have a clear migration path to the new Rectangle shape.

Current migration guide: https://bevyengine.org/learn/migration-guides/0-12-to-0-13/#deprecate-shapes-in-bevy-render-mesh-shape

## What solution would you like?

#10569 specifies that there should be a flipped method to cover this use case, however that seems to be missing.

```rs
let flipped_rectangle = Rectangle::new(100.0, 100.0).mesh().flipped();
```

**Comment by Jondolf** [comment]
This was sort of left out on purpose: https://github.com/bevyengine/bevy/pull/11431#issuecomment-1904776945

It seems a bit strange to me to support flipping a texture in a mesh representation. I feel like it's something that should instead be configurable at a material-level. I would assume that this is possible already? If not, I think it'd be valuable to have.

Only supporting texture flipping for `Rectangle`/`Quad` also seems slightly arbitrary to me; why not also support planes? Or circles? Or basically any shape?

Looking at other applications, I'm also not seeing a similar `flip` property elsewhere. For example, [Godot](https://docs.godotengine.org/en/stable/classes/class_quadmesh.html) doesn't have it (I also checked editor GUI), Unity doesn't have it, and Blender doesn't have it. To me, it doesn't feel like a quad-specific property, but a general material property.

**Comment by tripokey** [comment]
I'm not saying that the flipped method needs to be added to Mesh, as a user of `Quad::flip` I think that a line in the migration guide on how to achieve this in the new bevy release would be enough

**Comment by bugsweeper** [comment]
> I feel like it's something that should instead be configurable at a material-level

Do you mean just add to [StandardMaterial](https://docs.rs/bevy/latest/bevy/prelude/struct.StandardMaterial.html) additional parameter, or mention in migration guide that making custom flipping material is the correct way for replacement of `Quad::flip`?

**Comment by bugsweeper** [comment]
For any UV transformations you can use [StandardMaterial::uv_transform](https://github.com/janhohenheim/bevy/blob/ae64c5004e496f392cb9099850847489469f0d5d/crates/bevy_pbr/src/pbr_material.rs#L477), which is pretty fresh, because it had been added for [KHR_texture_transform support](https://github.com/bevyengine/bevy/pull/11904) to current main and it will be available from next 0.14 release.
For example for flip you can try set uv_transform to
`Affine2{matrix2: -Mat2::IDENTITY, translation: Vec2::ONE,}` for both axes flip 
or  
```
Affine2 {
    matrix2: Mat2::from_cols(-Vec2::X, Vec2::Y),
    translation: -Vec2::X + Vec2::Y,
} // for horizontal flip
```
instead of default or premultiply it with uv_transform that you wanted to use earlier, if so.
@alice-i-cecile How do you think, should we add flipped method to StandardMaterial, or maybe add FLIPPING / HORIZONTAL_FLIPPING const value to Affine2 consts?

**Comment by alice-i-cecile** [comment]
I think that `StandardMaterial` is the most predictable place to find this. I think we should do that, but *also* add some nice constants to `Affine2` (and possibly `Affine3`): those are great simple constants that will come up repeatedly and improve readability.
